packages/timeline-state-resolver/src/integrations/websocketTcpClient/$schemas/options.json

```diff
@@ -0,0 +1,43 @@
+{
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "title": "WebSocket & TCP Client Options",
+    "type": "object",
+    "properties": {
+      "webSocket": {
+        "type": "object",
+        "properties": {
+          "uri": {
+            "type": "string",
+            "ui:title": "WebSocket URI",
+            "ui:description": "URI to connect to, e.g. 'ws://localhost:8080'",
+            "description": "URI to connect to, e.g. 'ws://localhost:8080'"
+          },
+          "reconnectInterval": {
+            "type": "integer",
+            "ui:title": "Reconnect Interval",
+            "description": "Interval between reconnection attempts in milliseconds",
+            "default": 5000
+          }
+        },
+        "required": ["uri"]
+      },
+      "tcp": {
+        "type": "object", 
+        "properties": {
+          "host": {
+            "type": "string",
+            "ui:title": "Optional - TCP Host",
+            "description": "Optional TCP server host to connect to, e.g. '127.0.0.1' - some devices may have a TCP connection too"
+          },
+          "port": {
+            "type": "integer",
+            "ui:title": "TCP Port",
+            "description": "TCP server port to connect to"
+          }
+        },
+        "required": ["host", "port"]
```

Comment:
These host and port should be optional?

Reply:
Elsewhere it seems to be implied that the entire `tcp` object should be optional. I am not sure the Sofie UI will handle that well, and also it doesn't seem to be reflected in this JSON